src/extended_json.ts

```diff
@@ -36,7 +36,10 @@ type BSONType =
 
 export function isBSONType(value: unknown): value is BSONType {
   return (
-    isObjectLike(value) && Reflect.has(value, '_bsontype') && typeof value._bsontype === 'string'
+    value != null &&
+    typeof value === 'object' &&
+    '_bsontype' in value &&
```

Comment:
If you are wondering, yes, `'string' in x` does work even if the property is a getter